### PR TITLE
tests: Proto -> JSON conversion for roxctl export (except nodes)

### PR DESCRIPTION
**Pull request by rhybrillou**
### Description

The PR is part of a PR split stack.
See #11772 description for more information
PR stack:
- #11772
  - #11900 
    - #11901 
      - #11902 
        - #11903 
          - #11904 
            - #11905 

### User-facing documentation

(*must be* 2 items and both *must be* checked)

- [x] CHANGELOG update is not needed
- [x] Documentation is not needed

### Testing

- [x] inspected CI results

#### Automated testing

(*must be* at least 1 item and all items *must be* checked)

- [x] added unit tests
- [x] modified existing tests

#### How I validated my change

Manual run of `go test -v` in each directory where a test was either created or modified

### CODE REVIEW

File: roxctl/central/export/deployments/deployments_test.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	env.EXPECT().InputOutput().Times(1).Return(ioMock)
+
+	fakeCmd := &cobra.Command{}
+	flags.AddTimeoutWithDefault(fakeCmd, 10*time.Second)
+
+	cmd := Command(env)
+	err = cmd.RunE(fakeCmd, []string{})
+	assert.NoError(t, err)
+
+	var expectedDeploymentJSON = fixtures.GetExpectedJSONSerializedTestDeployment(t)
```
Comment (janisz) [inline roxctl/central/export/deployments/deployments_test.go:53]: This is used only here do we need to put it in a separated package? How about creating a json file and using go embed in test to load it?

File: roxctl/central/export/pods/pods_test.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	mockEnv.EXPECT().GRPCConnection().Times(1).Return(conn, nil)
+	mockEnv.EXPECT().InputOutput().Times(1).Return(mockIO)
+
+	fakeCmd := &cobra.Command{}
+	flags.AddTimeoutWithDefault(fakeCmd, 10*time.Second)
+
+	cmd := Command(mockEnv)
+	err = cmd.RunE(fakeCmd, []string{})
+	assert.NoError(t, err)
+	expectedSerializedPod := fixtures.GetExpectedJSONSerializedTestPod(t)
```
Comment (janisz) [inline roxctl/central/export/pods/pods_test.go:51]: ditto

File: roxctl/central/export/deployments/deployments_test.go (hunk 1 above)
Comment (rhybrillou) [reply roxctl/central/export/deployments/deployments_test.go]: I tried to have a structured approach to serialization tests before I split the PR. Alerts were used in multiple places in the codebase and I created a fixture in order to not duplicate the test object and associated serialized state across all tests. When I then moved to other objects types, I tried to reproduce the pattern.
I can and will then follow your suggestion for objects which are used in only one file.

**Review by janisz — APPROVED** [review]

**Pull request merged**